src/java.base/share/classes/java/lang/invoke/X-VarHandleSegmentView.java.template

```diff
@@ -33,723 +33,721 @@ import static java.lang.invoke.SegmentVarHandle.*;
+                bb.unsafeGetBase(),
+                offset(bb, base, offset));
 #else[byte]
-            return SCOPED_MEMORY_ACCESS.get$Type$Unaligned(bb.sessionImpl(),
-                    bb.unsafeGetBase(),
-                    offset(bb, base, offset),
-                    handle.be);
+        return SCOPED_MEMORY_ACCESS.get$Type$Unaligned(bb.sessionImpl(),
+                bb.unsafeGetBase(),
+                offset(bb, base, offset),
+                handle.be);
 #end[byte]
 #end[floatingPoint]
-        }
+    }
 
-        @ForceInline
-        static void set(VarHandle ob, Object obb, long base, $type$ value) {
-            set(ob, obb, base, ((SegmentVarHandle) ob).offset, value);
-        }
+    @ForceInline
+    static void set(VarHandle ob, Object obb, long base, $type$ value) {
+        set(ob, obb, base, ((SegmentVarHandle) ob).offset, value);
+    }
 
-        @ForceInline
-        static void set(VarHandle ob, Object obb, long base, long offset, $type$ value) {
-            SegmentVarHandle handle = (SegmentVarHandle)ob;
-            AbstractMemorySegmentImpl bb = handle.checkSegment(obb, base, false);
+    @ForceInline
+    static void set(VarHandle ob, Object obb, long base, long offset, $type$ value) {
+        SegmentVarHandle handle = (SegmentVarHandle)ob;
+        AbstractMemorySegmentImpl bb = handle.checkSegment(obb, base, false);
 #if[floatingPoint]
-            SCOPED_MEMORY_ACCESS.put$RawType$Unaligned(bb.sessionImpl(),
-                    bb.unsafeGetBase(),
-                    offset(bb, base, offset),
-                    $Type$.$type$ToRaw$RawType$Bits(value),
-                    handle.be);
+        SCOPED_MEMORY_ACCESS.put$RawType$Unaligned(bb.sessionImpl(),
+                bb.unsafeGetBase(),
+                offset(bb, base, offset),
+                $Type$.$type$ToRaw$RawType$Bits(value),
+                handle.be);
 #else[floatingPoint]
 #if[byte]
-            SCOPED_MEMORY_ACCESS.put$Type$(bb.sessionImpl(),
-                    bb.unsafeGetBase(),
-                    offset(bb, base, offset),
-                    value);
+        SCOPED_MEMORY_ACCESS.put$Type$(bb.sessionImpl(),
+                bb.unsafeGetBase(),
+                offset(bb, base, offset),
+                value);
 #else[byte]
-            SCOPED_MEMORY_ACCESS.put$Type$Unaligned(bb.sessionImpl(),
-                    bb.unsafeGetBase(),
-                    offset(bb, base, offset),
-                    value,
-                    handle.be);
+        SCOPED_MEMORY_ACCESS.put$Type$Unaligned(bb.sessionImpl(),
+                bb.unsafeGetBase(),
+                offset(bb, base, offset),
+                value,
+                handle.be);
 #end[byte]
 #end[floatingPoint]
-        }
-#if[!byte]
     }
+#if[!byte]
+}
 
-    static final class Aligned extends Unaligned {
+// This class must be accessed through non-aligned VarHandleSegmentAs$Type$s
+final class VarHandleSegmentAs$Type$sAligned extends VarHandleSegmentAs$Type$s {
```

Comment:
I kind of liked the nested classes -- why the change?

Reply:
Nesting means we need to define 3 classes: AsXxx, Unaligned, Aligned. I wish to reduce class loading by merging AsXxx and Unaligned. The Aligned still works as if it is nested, but I put it out to make the indentation beautiful for boolean and byte.